src/convertToObject.js

```diff
@@ -6,7 +6,24 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const entries = sourceString
+    .split(';')
+    .map((entry) => entry.trim())
+    .filter((entry) => entry);
+
+  const result = {};
+
+  for (const entry of entries) {
+    if (entry.includes(':')) {
+      const [rawKey, ...rawValueParts] = entry.split(':');
+      const key = rawKey.trim();
+      const value = rawValueParts.join(':').trim();
+
+      result[key] = value;
```

Comment:
Currently, if there are duplicate keys in the `sourceString`, the last value will overwrite any previous ones. If this is not the desired behavior, consider implementing logic to handle duplicate keys.